controllers/azuremachine_controller.go

```diff
@@ -109,215 +111,297 @@ func (r *AzureMachineReconciler) SetupWithManager(mgr ctrl.Manager, options cont
+    ctx, cancel := context.WithTimeout(context.Background(), reconciler.DefaultedLoopTimeout(r.ReconcileTimeout))
+    defer cancel()
+    logger := r.Log.WithValues("namespace", req.Namespace, "azureMachine", req.Name)
+
+    // Fetch the AzureMachine VM.
+    azureMachine := &infrav1.AzureMachine{}
+    err := r.Get(ctx, req.NamespacedName, azureMachine)
+    if err != nil {
+        if apierrors.IsNotFound(err) {
+            return reconcile.Result{}, nil
+        }
+        return reconcile.Result{}, err
+    }
+
+    // Fetch the Machine.
+    machine, err := util.GetOwnerMachine(ctx, r.Client, azureMachine.ObjectMeta)
+    if err != nil {
+        return reconcile.Result{}, err
+    }
+    if machine == nil {
+        r.Recorder.Eventf(azureMachine, corev1.EventTypeNormal, "Machine controller dependency not yet met", "Machine Controller has not yet set OwnerRef")
+        logger.Info("Machine Controller has not yet set OwnerRef")
+        return reconcile.Result{}, nil
+    }
+
+    logger = logger.WithValues("machine", machine.Name)
+
+    // Fetch the Cluster.
+    cluster, err := util.GetClusterFromMetadata(ctx, r.Client, machine.ObjectMeta)
+    if err != nil {
+        r.Recorder.Eventf(azureMachine, corev1.EventTypeNormal, "Unable to get cluster from metadata", "Machine is missing cluster label or cluster does not exist")
+        logger.Info("Machine is missing cluster label or cluster does not exist")
+        return reconcile.Result{}, nil
+    }
+
+    logger = logger.WithValues("cluster", cluster.Name)
+
+    // Return early if the object or Cluster is paused.
+    if annotations.IsPaused(cluster, azureMachine) {
+        logger.Info("AzureMachine or linked Cluster is marked as paused. Won't reconcile")
+        return ctrl.Result{}, nil
+    }
+
+    azureClusterName := client.ObjectKey{
+        Namespace: azureMachine.Namespace,
+        Name:      cluster.Spec.InfrastructureRef.Name,
+    }
+    azureCluster := &infrav1.AzureCluster{}
+    if err := r.Client.Get(ctx, azureClusterName, azureCluster); err != nil {
+        r.Recorder.Eventf(azureMachine, corev1.EventTypeNormal, "AzureCluster unavailable", "AzureCluster is not available yet")
+        logger.Info("AzureCluster is not available yet")
+        return reconcile.Result{}, nil
+    }
+
+    logger = logger.WithValues("AzureCluster", azureCluster.Name)
+
+    // Create the cluster scope
+    clusterScope, err := scope.NewClusterScope(scope.ClusterScopeParams{
+        Client:       r.Client,
+        Logger:       logger,
+        Cluster:      cluster,
+        AzureCluster: azureCluster,
+    })
+    if err != nil {
+        r.Recorder.Eventf(azureCluster, corev1.EventTypeWarning, "Error creating the cluster scope", err.Error())
+        return reconcile.Result{}, err
+    }
+
+    //Fetch the AzureIPPool
+    azureIPPoolName := client.ObjectKey{
+        Namespace: "default",
+        Name:      "ase-ip-pool", //get the ip-pool-name from the cluster spec
```

Comment:
should this be hardcoded?

Reply:
For now there is only going to be one IP pool object and we form the input yaml for it in ASE code. So we can have a default name.